### PR TITLE
Minor improvements to Annotations Tutorial

### PR DESCRIPTION
**Pull request by timhoffm**
Follow up to #23606.

- Move the basic text() vs. annotate() discussion to the top
- Styling in "Annotating with boxed text": move the text from Axes edge to center; less glaring colors

There's still more that can be done in later PRs ...

### CODE REVIEW

File: tutorials/text/annotations.py [hunk 1]
```diff
@@ -2,7 +2,17 @@
...
+add a text with `~.Axes.text`, this is usually not convenient, because
+the text should usually not be placed exactly at the data point, requiring
+manual calculation of the coordinates.
+
+`~.Axes.annotate` makes annotations easier because
+
+- it uses explicit coordinates for both the text and the data point. This
+  makes placing the text in a desired position easier.
+- it allows adding an arrow from the text to the data point. The arrow can be
+  styled in various ways.
```
[the remark this answers is not in the document]
Comment (jklymak) [reply tutorials/text/annotations.py:15]: > as it is intended for plotting text as data.  

I'd leave opinions like this out.  I use `ax.text` to annotate data and axes all the time.  I only use `ax.annotate` if I need an arrow, as the interface is far clunkier.  

I also think we should differentiate between "annotating" and "annotations", which is a whole class of artists (`axv/hline`, 'fill_between/x`, `axv/hspan`, `text`, `annotate`, probably others) and `ax.annotate` which is a single annotation method.  I'm actually not a fan of this tutorial squatting on the name "annotations".

**Comment by timhoffm** [comment]
@story645 @jklymak I've just taken over the suggestions. I'm not up to fine-tuning the semantics of "annotation" or whether or not `text()` is an option for simple annotations. Please do that in follow-up PRs, or speak up to take over this PR push further improvements.

File: tutorials/text/annotations.py [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
 
-Annotating text with Matplotlib.
+Annotations are graphical elements, often pieces of text, that explain, add
+context to, or otherwise highlight some portion of the visualized data.
+`~.Axes.annotate` supports a number of coordinate systems for flexibly
+positioning data and annotations relative to each other and a variety of
+options of for styling the text. Axes.annotate also provides an optional arrow
+from the text to the data and this arrow can be styled in various ways.
+`~.Axes.text` can also be used for simple text annotation, but does not
+provide flexibility in positioning as it is intended for plotting text as data.
```
[the remark this answers is not in the document]
Comment (timhoffm) [reply tutorials/text/annotations.py]: Yes.

**Comment by timhoffm** [comment]
I self-merge based on @story645's approval.

**Pull request merged**